File: 2395-find-subarrays-with-equal-sum/2395-find-subarrays-with-equal-sum.py
class Solution:
    def findSubarrays(self, nums: List[int]) -> bool:
        seen = set()
        for i in range(1, len(nums)):
            s = nums[i] + nums[i - 1]
            if s in seen:
                return True
            seen.add(s)
        return False

File: 2395-find-subarrays-with-equal-sum/2395-find-subarrays-with-equal-sum.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var findSubarrays = function(nums) {
  const set = new Set();
  
  for (let i = 1; i < nums.length; i++) {
    const sum = nums[i] + nums[i - 1];
    if (set.has(sum)) return true;
    
    set.add(sum);
  }
  
  return false;
};